src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java

```diff
@@ -93,10 +98,30 @@ public void saveModifications(UUID groupUuid, List<? extends ModificationEntity>
         var modificationGroupEntity = this.modificationGroupRepository
                 .findById(groupUuid)
                 .orElseGet(() -> modificationGroupRepository.save(new ModificationGroupEntity(groupUuid)));
-        modifications.forEach(m -> m.setGroup(modificationGroupEntity));
+        modificationGroupEntity.getModifications().addAll(modifications);
         this.modificationRepository.saveAll(modifications);
     }
 
+    @Transactional // To have all move in the same transaction (atomic)
+    public void moveModifications(UUID groupUuid, List<UUID> modifications, UUID before) {
+        var modificationGroupEntity = getModificationGroup(groupUuid);
+
+        Map<UUID, ModificationEntity> originalModifications = modificationRepository.findAllBaseByGroupId(groupUuid).stream()
+            .collect(Collectors.toMap(ModificationEntity::getId, Function.identity(), (x, y) -> y, LinkedHashMap::new));
+
+        if (!originalModifications.keySet().containsAll(modifications) || (before != null && !originalModifications.containsKey(before))) {
```

Comment:
`before` can be null ?
Perhaps `@NonNull`

Reply:
before can be null (will add comment)